### DISCUSSION TITLE
Which gamepad do you recommend in 2022??

### DISCUSSION THREAD

**Discussion by bacodes** [bitcraze/discussions#219] [Q&A]
Hi,

I want to start with the crazyflie 2.1 and I can't find any new information about the currently best gamepad.
There are just some unofficial tips about the old gamepads from Sony for ps3 or ps4 or from 8BitDo. But this information is also years old (https://forum.bitcraze.io/viewtopic.php?p=21267&sid=0c1a64bffa969f82704cf8a8118248fd#p21267).
Even the iOS-app is just for Crazyflie 2.0. That seems too old!
Aren't there any updates about how to controle the drone or do the old gamepads and iOS-app still work?

Which gamepad do you recommend in 2022 for controlling Crazyflie 2.1 and where can you find the corresponding installation instructions?

Thank you.

**Comment by tobbeanton** [comment]
The cfclient contains some [gamepad documentation](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/inputdevices/).

I would recommend a high quality gamepad, the cheap ones often doesn't have good resolution and makes the crazyflie harder to control. Game console gamepads are generally high quality such as Playstation and Xbox.

**Reply by bacodes** [reply]
Thank you. The gamepad documentation ist old, too (it is for Windows 7 & 8). 
Do you think I can use a new Sony Gamepad like the one for PS5?
Or when I use an old one like for PS3 does it need to have dual shock or mustn't have dual shock or doesn't that matter?
Isn't there anything else to consider by choosing a gamepad?

**Reply by knmcguire** [reply]
If your operating system is able to retrieve the data from any gamepad, from which you can google it it can, you could create a new gamepad mapping for it via these instructions. https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/#create-new-mappings

We haven't used any controller newer than either an xbox360 or ps3, but I'm sure that will work too.

**Reply by bacodes** [reply]
Hello Kimberley,
Thank you for your answer. But I have to say it makes me really sad. What you wrote means to me that the crazyflie project is dying. The entry hurdle is not as low as it can be, it is actually far from it. And that's not a good way to sell a product line. And the Crazyflie project can only thrive if as many newcomers as possible join it. Otherwise, purely commercial drone providers will soon overtake Crazyflie in the research sector as well. And if you don't even test current gamepads, let alone have current documentation easily accessible, if you don't also offer Crazyflyie with a start-out-of-the-box option, then that's it for this promising research project, because it just keeps getting better pushed into its niche and will not develop significantly further.

I would recommend always providing up-to-date gamepads or at least up-to-date, dated info on how to start Crazyflie. And if years-old information is still up-to-date, then this must be made clear in updated descriptions.
Also, I would recommend that you come up with a concept that provides a guideline on how to program with Crazyflie in Python. Also especially for beginners. They have to be won for the Crazyflie project. Specialists then come on their own, you don't have to worry about them.

**Reply by knmcguire** [reply]
Sorry if it came across like that. The usecases have shifted in the last years from manual flight to  autonomous flight and personally I hardly even use a controller myself. If I want to fly the crazyflie, I either use the app for manual flight or [the commander based flight control panel](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/flightcontrol_tab/) I just want the have a quick test out of the flight performance with positioning systems. Feeling is most people do this too as we haven't had any contributions to our opensource repositories for mapping of newer controllers for a while we will be happy to accept pull requests from those if somebody feels the need. Most newer contribution has mostly been on the improvement of the onboard autonomy of the crazyflie and improvement of the positioning system quality so that is what we are currently focusing our efforts on.

Anyway, if you need more help on your project just let us know on the discussions and we will point you to the right place in the [documentation](https://www.bitcraze.io/documentation/system/).

**Comment by knmcguire** [comment]
I've made an issue on the cfclient repo about your request for newer controllers: https://github.com/bitcraze/crazyflie-clients-python/issues/627 so maybe somebody will pick that up from the community if there is more interest outside [of mapping it manually](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/#create-new-mappings)

The IOS app and the android app still work also for the crazyflie 2.1 (the text just hasn't been updated) as that relies on the same communication protocol. Currently we don't get enough request to update them so as long as they still work (which we test at every release), we will probably not update them with new functionality any time soon.
